tests/pass-dep/libc/libc-epoll.rs

```diff
@@ -270,3 +271,23 @@ fn test_epoll_eventfd() {
     let expected_value = u64::try_from(fd).unwrap();
     assert!(check_epoll_wait::<8>(epfd, vec![(expected_event, expected_value)]));
 }
+
+fn test_pointer() {
+    // Create an epoll instance.
+    let epfd = unsafe { libc::epoll_create1(0) };
+    assert_ne!(epfd, -1);
+
+    // Create a socketpair instance.
+    let mut fds = [-1, -1];
+    let res = unsafe { libc::socketpair(libc::AF_UNIX, libc::SOCK_STREAM, 0, fds.as_mut_ptr()) };
+    assert_eq!(res, 0);
+
+    // Register fd[1] with EPOLLIN|EPOLLOUT|EPOLLET
+    // EPOLLET is negative number for i32 so casting is needed to do proper bitwise OR for u32.
+    let epollet = libc::EPOLLET as u32;
+    let flags = u32::try_from(libc::EPOLLIN | libc::EPOLLOUT | libc::EPOLLRDHUP).unwrap() | epollet;
+    let data = MaybeUninit::<u64>::uninit().as_ptr();
+    let mut ev = libc::epoll_event { events: u32::try_from(flags).unwrap(), u64: data as u64 };
```

Comment:
```suggestion
    let mut ev = libc::epoll_event { events: u32::try_from(flags).unwrap(), u64: data.expose_provenance() as u64 };
```

Reply:
``expose_provenance`` is on nightly so error will be thrown when it is used? Is there any way to get around this? 

Error:
```bash
error[E0658]: use of unstable library feature 'exposed_provenance'
   --> ./tests/pass-dep/libc/libc-epoll.rs:292:19
    |
292 |         u64: data.expose_provenance() as u64,
    |                   ^^^^^^^^^^^^^^^^^
    |
    = note: see issue #95228 <https://github.com/rust-lang/rust/issues/95228> for more information
    = help: add `#![feature(exposed_provenance)]` to the crate attributes to enable
    = note: this compiler was built on 2024-07-06; consider upgrading it if it is out of date

error: aborting due to 1 previous error
```

Reply:
you can add the feature gate